src/ArcxAnalyticsSdk.ts

```diff
@@ -201,12 +216,18 @@ export class ArcxAnalyticsSdk {
     https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendtransaction
   */
   private _trackTransactions(): boolean {
-    const provider = getWeb3Provider()
+    const provider = this.provider
     if (!provider) {
       return false
     }
 
-    const request = provider.request
+    if (!this._originalRequest) {
+      this._originalRequest = provider.request
+    }
+
+    // Deliberately not using this._original request to not intefere with the signature tracking's
+    // request modification
+    const request = provider.request.bind(provider)
```

Comment:
Without `.bind(provider)`, the request call fails as it loses its link to the provider